src/TurbulenceClosures/turbulence_closure_implementations/Smagorinskys/scale_invariant_operators.jl

```diff
@@ -0,0 +1,143 @@
+using Oceananigans.TurbulenceClosures: tr_Σ, Σ₁₁, Σ₂₂, Σ₃₃, Σ₁₂, Σ₁₃, Σ₂₃         
+using Oceananigans.TurbulenceClosures: tr_Σ², Σ₁₂², Σ₁₃², Σ₂₃²         
+using Oceananigans.Operators: volume
+
+#####
+##### Filters
+#####
+
+const AG{FT} = AbstractGrid{FT} where FT
+
+# For a regularly-spaced grids, the double interpolations below
+@inline ℱ²ᵟ(i, j, k, grid::AG{FT}, f::Field{Face,   Center, Center}) where FT = ℑxyzᶠᶜᶜ(i, j, k, grid, ℑxyzᶜᶠᶠ, f)
+@inline ℱ²ᵟ(i, j, k, grid::AG{FT}, f::Field{Center, Face,   Center}) where FT = ℑxyzᶜᶠᶜ(i, j, k, grid, ℑxyzᶠᶜᶠ, f)
+@inline ℱ²ᵟ(i, j, k, grid::AG{FT}, f::Field{Center, Center, Face  }) where FT = ℑxyzᶜᶜᶠ(i, j, k, grid, ℑxyzᶠᶠᶜ, f)
+@inline ℱ²ᵟ(i, j, k, grid::AG{FT}, f::Field{Center, Center, Center}) where FT = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, f)
+@inline ℱ²ᵟ(i, j, k, grid::AG{FT}, f::F, args...) where {FT, F <:Function} = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, f, args...)
```

Comment:
```suggestion
@inline ℱ²ᵟ(i, j, k, grid::AG, f::Field{Face,   Center, Center}) = ℑxyzᶠᶜᶜ(i, j, k, grid, ℑxyzᶜᶠᶠ, f)
@inline ℱ²ᵟ(i, j, k, grid::AG, f::Field{Center, Face,   Center}) = ℑxyzᶜᶠᶜ(i, j, k, grid, ℑxyzᶠᶜᶠ, f)
@inline ℱ²ᵟ(i, j, k, grid::AG, f::Field{Center, Center, Face  }) = ℑxyzᶜᶜᶠ(i, j, k, grid, ℑxyzᶠᶠᶜ, f)
@inline ℱ²ᵟ(i, j, k, grid::AG, f::Field{Center, Center, Center}) = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, f)
@inline ℱ²ᵟ(i, j, k, grid::AG, f::Function, args...) = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, f, args...)
```
why for the function it is assumed the ccc location?

Reply:
Hmm, it won't work anyways to use `Field` in a kernel because it is adapted for GPU.

Reply:
I fixed it

Reply:
i fixed this